l10n_it_fiscalcode/model/res_partner.py

```diff
@@ -6,30 +7,46 @@
 from odoo.exceptions import ValidationError
 
 
+def _check_company_fiscal_code(fiscal_code):
+    """`fiscal_code` is the Fiscal Code of a Company."""
+    return len(fiscal_code) == 11
+
+
+def _check_person_fiscal_code(fiscal_code):
+    """`fiscal_code` is the Fiscal Code of a Person."""
+    return len(fiscal_code) == 16
+
+
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
     @api.constrains("fiscalcode")
```

Comment:
Non ci vorrebbe un controllo anche nel caso cambiasse `company_type`?

In pratica dovrebbe essere possibile cambiare solo con:
```
partner.write({"company_type": whatever, "fiscalcode": False})
partner.write({"company_type": "person", "fiscalcode": string_of_len16})
partner.write({"company_type": "company", "fiscalcode": string_of_len11})
```

Reply:
> Non ci vorrebbe un controllo anche nel caso cambiasse `company_type`?

È stato aggiunto da poco in https://github.com/OCA/l10n-italy/commit/eb161c7243ca474aca5d21c1083faed16d492612.